src/main/groovy/lang/ObjectRange.java

```diff
@@ -306,14 +303,13 @@ public int size() {
                 }
             } else {
                 // let's brute-force calculate the size by iterating start to end
-                final Iterator iter = new StepIterator(this, 1);
```

Comment:
synchronization serves as an inlining barrier, thus I do not agree to it being cheap in all cases. If not needed it should be avoided.